contract/src/main/java/io/pravega/schemaregistry/contract/data/Compatibility.java

```diff
@@ -0,0 +1,203 @@
+import lombok.Builder;
+import lombok.Data;
+
+/**
+ * Defines different Compatibility policy options for schema evolution for schemas within a group.
+ * The choice of compatibility policy tells the Schema Registry service whether a schema should be accepted to evolve
+ * into new schema by comparing it with one or more existing versions of the schema. 
+ * 
+ * {@link Type#AllowAny}: allow any changes to schema without any checks performed by the registry. 
+ * {@link Type#DenyAll}: disables any changes to the schema for the group.
+ * {@link Type#Backward}: a new schema can be used to read data written by previous schema. 
+ * {@link Type#BackwardTransitive}: a new schema can be used read data written by any of previous schemas. 
+ * {@link Type#BackwardTill}: a new schema can be used to read data written by any of previous schemas till schema 
+ * identified by version {@link Compatibility#backwardTill}. 
+ * {@link Type#Forward}: previous schema can be used to read data written by new schema. 
+ * {@link Type#ForwardTransitive}: all previous schemas can read data written by new schema. 
+ * {@link Type#ForwardTill}: All previous schemas till schema identified by version {@link Compatibility#forwardTill}
+ * can read data written by new schema. 
+ * {@link Type#Full}: both backward and forward compatibility.
+ * {@link Type#FullTransitive}: both backward and forward compatibility with all previous schemas.
+ * {@link Type#BackwardAndForwardTill}: All previous schemas till schema identified by version {@link Compatibility#forwardTill}
+ * can read data written by new schema. New schema can be used to read data written by any of previous schemas till schema 
+ * identified by version {@link Compatibility#backwardTill}. 
+ */
+@Data
+@Builder
+public class Compatibility implements SchemaValidationRule {
+    /**
+     * Enum that defines the Type of compatibility policy.
+     */
+    private final Type compatibility;
+    /**
+     * Version info to be specified if the compatibility policy choic.e is either {@link Type#backwardTill} or 
+     * {@link Type#backwardTillAndForwardTill}.
+     */
+    private final VersionInfo backwardTill;
+    /**
+     * Version info to be specified if the compatibility policy choice is either {@link Type#forwardTill} or 
+     * {@link Type#backwardTillAndForwardTill}.
+     */
+    private final VersionInfo forwardTill;
+
+    private Compatibility(Type compatibility) {
+        this(compatibility, null, null);
+    }
+
+    public Compatibility(Type compatibility, VersionInfo backwardTill, VersionInfo forwardTill) {
+        this.compatibility = compatibility;
+        this.backwardTill = backwardTill;
+        this.forwardTill = forwardTill;
+    }
+
+    @Override
+    public String getName() {
+        return Compatibility.class.getSimpleName();
+    }
+
+    public enum Type {
+        AllowAny,
+        DenyAll,
+        Backward,
+        BackwardTill,
+        BackwardTransitive,
```

Comment:
tkaitchuck 5 days ago Member
```
Can "BackwardTransitive" be made into a special case of "Backward" where the version is null? (Same for forward)
```
@shiveshr
shiveshr 5 days ago Author Collaborator
```
We can make it a special case. Though that reduces the number of values in the enum, that probably increases the complexity in expressing the values for the user.
```
@tkaitchuck
tkaitchuck 3 days ago Member
```
I think there may be a way to get the best of both approaches.
If we define a builder for Compatibility that has setters for the actual config values it can be documented and give users fine grained control. But keep the static method constructors below for each of these common combinations. This keeps the state simple but allows the common configuations to be used quickly and easily.
```
@shiveshr
shiveshr 3 days ago Author Collaborator
```
The thing is, we can define that builder on our client library. But service is exposed as a REST endpoint.
IDK how we could extend such constructors and builders and guide users on the meaningful values.
Therefore including all meaningful values into an enum and exposing it directly to the user may be the simpler approach.
```

Reply:
>The thing is, we can define that builder on our client library. But service is exposed as a REST endpoint. IDK how we could extend such constructors and builders and guide users on the meaningful values.

We would not. This is precisely my intention. See my comment above.